server/portal/views/base.py

```diff
@@ -27,52 +26,62 @@ def dispatch(self, request, *args, **kwargs):
+            logger.error(e, exc_info=True)
+            raise e
+        except ApiException as e:
+            status = e.response.status_code
+            message = e.response.reason
+            extra = e.extra
             if status != 404:
                 logger.error(
                     '%s: %s',
                     message,
-                    err.response.text,
+                    e.response.text,
                     exc_info=True,
                     extra=extra
                 )
             else:
                 logger.info('Error %s', message, exc_info=True, extra=extra)
             return JsonResponse({'message': message}, status=400)
-        except (ConnectionError, HTTPError) as err:
-            status = err.response.status_code
-            content = err.response.json()
-            message = content.get("message", "Unknown Error")
-            if status in [404, 403]:
-                logger.warning(
-                    '%s: %s',
-                    message,
-                    err.response.text,
-                    exc_info=True,
-                    extra={
-                        'username': request.user.username,
-                        'session_key': request.session.session_key
-                    }
-                )
+        except (ConnectionError, HTTPError) as e:
+            if e.response is not None:
+                status = e.response.status_code
+                try:
+                    content = e.response.json()
+                    message = content.get("message", "Unknown Error")
+                except ValueError:
+                    message = "Unknown Error"
+                if status in [404, 403]:
+                    logger.warning(
+                        '%s: %s',
+                        message,
+                        e.response.text,
+                        exc_info=True,
+                        extra={
+                            'username': request.user.username,
+                            'session_key': request.session.session_key
+                        }
+                    )
+                else:
+                    logger.error(
+                        '%s: %s',
+                        message,
+                        e.response.text,
+                        exc_info=True,
+                        extra={
+                            'username': request.user.username,
+                            'session_key': request.session.session_key
+                        }
+                    )
             else:
-                logger.error(
-                    '%s: %s',
-                    message,
-                    err.response.text,
-                    exc_info=True,
-                    extra={
-                        'username': request.user.username,
-                        'session_key': request.session.session_key
-                    }
-                )
+                logger.error('{}'.format(e), exc_info=True)
```

Comment:
Could we do one of `logger.error'({}.format(e))` or `logger.error(e)`?

Reply:
👍   just to confirm, you're wanting to drop full stack trace?

Reply:
No, asking if we could do one or the other - what's the difference? Once spits out the full stacktrace and the other does not?

Reply:
yeah, with the exec_info, they spit out the full stack strace:
`logger.error(e, exc_info=True)` or  `logger.error'({}.format(e), exc_info=True)`
so basically same as:
`logger.exception(e)`

I think I might have misunderstood the original comment.  are you asking about being consistent with all logger.error statements in this file?  like not have different lines logging the exact same thing but using different syntax (like `logger.error('{}'.format(e), exc_info=True`) prints the exact same as `logger.error(e, exc_info=True)`)

Reply:
> are you asking about being consistent with all logger.error statements in this file? like not have different lines logging the exact same thing but using different syntax (like logger.error('{}'.format(e), exc_info=True) prints the exact same as logger.error(e, exc_info=True))

yes, exactly - sorry for being vague

Reply:
👍 , i'll do some housekeeping.